feat(linter): support `env` and `globals` in `overrides` configuration

Sysix opened the pull request:
This PR supports `env` and `globals` keys in `overrides[x]`.

I added tests where I could. Snapshot CLI tests are working 🥳

Sysix commented:
* **#8915** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8915?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8915?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8915).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented on crates/oxc_linter/src/config/env.rs:
```diff
@@ -70,4 +75,17 @@ mod test {
     }
+
+    #[test]
+    fn test_override_envs() {
```
might be good to add some more tests around this, though probably not critical since we have the end-to-end tests via the oxlintrc files. was thinking we could add a test for checking that you can: 1) replace the value (true -> false), 2) add a key (empty -> 1 key)

Sysix replied:
I implemented the test in another place: https://github.com/oxc-project/oxc/pull/8915/files#diff-4803bbea196e86eff70b44ffa5fef5e38573bfeeaf90cb578cba00b20e2694acR302-R405
This fitted more because the overrides plugin/rules tests are here too

camchenry commented on crates/oxc_linter/src/config/globals.rs:
```diff
@@ -174,4 +180,20 @@ mod test {
     }
+
+    #[test]
+    fn test_override_globals() {
```
same comment as above about adding tests, but again probably not critical

camchenry commented on crates/oxc_linter/src/config/overrides.rs:
```diff
@@ -201,4 +205,44 @@ mod test {
+        }))
+        .unwrap();
+
+        assert!(config.env.clone().unwrap().contains("es2022"));
```
do we need to clone here? could we just use a reference to `env`?

camchenry reviewed (approved):
❤️ 

some small test suggestions, but feel free to merge and address later as you see fit

Boshen commented:
### Merge activity

* **Feb 9, 7:40 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(linter): support `env` and `globals` in `overrides` configuration (#8915)

This PR supports `env` and `globals` keys in `overrides[x]`.

I added tests where I could. Snapshot CLI tests are working 🥳
